e107_plugins/forum/shortcodes/batch/forum_shortcodes.php

```diff
@@ -658,6 +671,20 @@ function sc_lastpost()
         return $this->lastpostdata('post');
 	}
 
+	function sc_lastpostdateonly()
+	{
+        return $this->lastpostdata('dateonly');
+	}
+
+	function sc_lastpostlink()
+	{
+        return $this->lastpostdata('link');
+	}
+
+	function sc_lastpostname()
+	{
+        return $this->lastpostdata('name');
+	}

```

Comment:
This is why we have $parms, to avoid situations like this..  lastpostxxxxx()
What we need here is sc_lastpost($parm)
check for $parm['type'] == 'link' etc.